src/main/java/com/example/controller/GraphController.java

```diff
@@ -196,6 +212,130 @@ public void initGraphView() {
+        DocumentBuilder builder = factory.newDocumentBuilder();
+        Document doc = builder.parse(file);
+
+        Map<String, String> keys = new HashMap<>();
+        List<MyVertex<Integer>> vertices = new LinkedList<>();
+        List<Pair<Integer, Integer>> edges = new LinkedList<>();
+
+        // Get data keys
+        NodeList keyNodes = doc.getElementsByTagName("key");
+        for (int i = 0; i < keyNodes.getLength(); i++) {
+            org.w3c.dom.Node node = keyNodes.item(i);
+            String id = node.getAttributes().getNamedItem("id").getNodeValue();
+            keys.put(id,
+                    node.getAttributes().getNamedItem("attr.name").getNodeValue());
+        }
+
+        // Get vertices
+        NodeList vertexNodes = doc.getElementsByTagName("node");
+        for (int i = 0; i < vertexNodes.getLength(); i++) {
+            org.w3c.dom.Node node = vertexNodes.item(i);
+            MyVertex<Integer> newVertex = new MyVertex<>(Integer.parseInt(node.getAttributes().getNamedItem("id").getNodeValue()));
+            boolean traitor = false; //todo default value from graphML
+            boolean opinion = false;
+
+            NodeList dataNodes = node.getChildNodes();
+            for (int j = 0; j < dataNodes.getLength(); j++) {
+                org.w3c.dom.Node dataNode = dataNodes.item(j);
+                String key = dataNode.getAttributes().getNamedItem("key").getNodeValue();
+                String attribute = keys.get(key);
+                if (key != null) {
+                    switch (attribute) {
+                        case "traitor" -> traitor = Boolean.parseBoolean(dataNode.getChildNodes().item(0).getNodeValue());
+                        case "attack" -> opinion = Boolean.parseBoolean(dataNode.getChildNodes().item(0).getNodeValue());
+                    }
+
+                }
+            }
+            newVertex.setIsTraitor(traitor);
+            newVertex.setIsSupporting(opinion);
+            vertices.add(newVertex);
+        }
+
+        //Get edges
+        NodeList edgeNodes = doc.getElementsByTagName("edge");
+        for (int i = 0; i < edgeNodes.getLength(); i++) {
+            org.w3c.dom.Node node = edgeNodes.item(i);
+            int from = Integer.parseInt(node.getAttributes().getNamedItem("source").getNodeValue());
+            int to = Integer.parseInt(node.getAttributes().getNamedItem("target").getNodeValue());
+            edges.add(new Pair<>(from, to));
+        }
+
+        //Reset graph
+        MyGraph<Integer, Integer> newGraph = new MyGraph<>();
+        int newCounter = 0;
+        for (MyVertex<Integer> vertex : vertices) {
+            newGraph.insertVertex(vertex);
+            vertexIdCounter = Integer.max(vertex.element(), vertexIdCounter);
+        }
+
+        for (Pair<Integer, Integer> edge : edges) {
+            newGraph.insertEdge(edge.getKey(), edge.getValue(), 1);
+        }
+
+        this.setModelGraph(newGraph);
+        vertexIdCounter = newCounter;
```

Comment:
If we reset that counter, there will be an error while adding new vertex. It has to be removed.